compiler/testData/diagnostics/testsWithJvmBackend/inlineCycle.kt

```diff
@@ -1,5 +1,5 @@
 // !RENDER_DIAGNOSTICS_FULL_TEXT
-
+// TARGET_BACKEND: JVM_OLD
```

Comment:
Please rebase on master, as I remember @dnpetrov supported cycle check in IR

Reply:
This is rebased on master. The change in this test is that both functions used to be compiled twice, thus the INLINE_CALL_CYCLE is also generated twice on each call that is part of the cycle. With the patch, there is only one INLINE_CALL_CYCLE per call. This only affects the test and not user-visible diagnostics (inlineCycle.txt and inlineCycle_ir.txt are the same) because they are deduplicated before printing.

Reply:
By the way, I had to reimplement inline cycle checking without using the old backend code because it's delayed by one level of nesting. Consider the test:
```
inline fun f() { g() }
inline fun g() { f() }
```
What used to happen here is that `f` pushed `g` to a stack in `enterIntoInlining`, then `g` pushed `f` to the same stack, and finally `f` tried to push `g` again and noticed that `g` is already on the stack. This, however, requires starting bytecode generation for `f` *twice*, since it's the second pass that detects the cycle. With my patch, this is not possible: when `g` tries to inline `f`, it gets an incomplete method node from the class codegen, and the diagnostics have not been generated yet. This is why there is now a new stack in `JvmBackendContext` which codegen for `g` can look into to emit the diagnostics when it sees the incomplete node.

Reply:
Could we make common solution for both backends with remembering original compilation context (function `f`) in same stack so delay would be avoided?

Reply:
Maybe. I'll try and submit that as a separate PR if possible.

Reply:
Done in #3194